src/main/java/application/controllers/GraphController.java

```diff
@@ -48,20 +48,54 @@ public GraphController(MainController m) {
         this.getRoot().setVbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
 
         this.getRoot().addEventFilter(ScrollEvent.SCROLL, event -> {
+            if (graphMouseHandling.getPrevClick() != null) {
+                graphMouseHandling.getPrevClick().resetFocus();
+            }
             if (event.getDeltaY() != 0) {
+                if (graphMouseHandling.getPrevClick() != null) {
+                    graphMouseHandling.getPrevClick().resetFocus();
+                }
                 if (event.getDeltaY() < 0) {
                     mainController.switchScene(+1);
+                    if (graphMouseHandling.getPrevClick() != null) {
+                        focus(graphMouseHandling.getPrevClick());
+                    }
                     event.consume();
-
-                } else if (event.getDeltaY() > 0) {
+                }
+                if (event.getDeltaY() > 0) {
                     mainController.switchScene(-1);
+                    if (graphMouseHandling.getPrevClick() != null) {
+                        focus(graphMouseHandling.getPrevClick());
+                    }
                     event.consume();
                 }
             }
         });
     }
 
     /**
+     * Method to focus on a Cell.
+     *
+     * @param prevClick the cell to focus to.
+     */
+    public void focus(Cell prevClick) {
+        prevClick.resetFocus();
+        for (Cell c : graph.getModel().getAllCells()) {
+            if (c.getCellId() == prevClick.getCellId()) {
+                prevClick = c;
+                break;
+            }
+            if (c.getCellId() > prevClick.getCellId()) {
+                prevClick = c;
+                break;
+            }
+        }
```

Comment:
can be simplified to 
if ((c.getCellId() == prevClick.getCellId()) || (c.getCellId() > prevClick.getCellId())) {
           prevClick = c;
           break;
 }